diff --git a/dist/config/adapter.js b/dist/config/adapter.js
index ea1d21db02d413de0d25367c646cd9773fb41223..42aed198ffb18752a7a245e1f8a0f2bf845b80d5 100644
--- a/dist/config/adapter.js
+++ b/dist/config/adapter.js
@@ -16,6 +16,7 @@ const getRawOptions = (options, compiler, processResource) => {
     const mode = options.mode;
     const experiments = getRawExperiments(options.experiments);
     return {
+        features: options.features,
         mode,
         target: getRawTarget(options.target),
         context: options.context,
diff --git a/dist/config/defaults.js b/dist/config/defaults.js
index b413bb21a053a161a64056664354a50ee6429283..9983bc69e39d031ca49fa04bcad0aea093e4b996 100644
--- a/dist/config/defaults.js
+++ b/dist/config/defaults.js
@@ -53,6 +53,11 @@ const applyRspackOptionsDefaults = (options) => {
     applyExperimentsDefaults(options.experiments, {
         cache: options.cache
     });
+    if (options.features) {
+        applyFeaturesDefaults(options.features);
+    } else {
+        D(options, 'features', {});
+    }
     applySnapshotDefaults(options.snapshot, { production });
     applyModuleDefaults(options.module, {
         // syncWebAssembly: options.experiments.syncWebAssembly,
@@ -129,6 +134,13 @@ const applyExperimentsDefaults = (experiments, { cache }) => {
         D(experiments.rspackFuture, "disableApplyEntryLazily", false);
     }
 };
+const applyFeaturesDefaults = (features) => {
+    D(features, 'split_chunks_strategy', {});
+    if (typeof features.split_chunks_strategy === 'object') {
+        D(features.split_chunks_strategy, 'name', '');
+        D(features.split_chunks_strategy, 'topLevelFrameworks', []);
+    }
+};
 const applySnapshotDefaults = (snapshot, { production }) => {
     F(snapshot, "module", () => production
         ? { timestamp: true, hash: true }
diff --git a/dist/config/normalization.js b/dist/config/normalization.js
index 7769ba679c50a63a60961cfe87f5b7b950593e85..d159573dacd63aec503be254366eea94cdddf826 100644
--- a/dist/config/normalization.js
+++ b/dist/config/normalization.js
@@ -13,6 +13,7 @@ exports.getNormalizedRspackOptions = void 0;
 const util_1 = require("../util");
 const getNormalizedRspackOptions = (config) => {
     return {
+        features: config.features,
         ignoreWarnings: config.ignoreWarnings !== undefined
             ? config.ignoreWarnings.map(ignore => {
                 if (typeof ignore === "function") {
diff --git a/dist/config/zod.js b/dist/config/zod.js
index 9efb9502f22278229c9c5798b7ae5eee2dbd0915..348b9528161995d65d462279f74960d1b00e195f 100644
--- a/dist/config/zod.js
+++ b/dist/config/zod.js
@@ -759,6 +759,7 @@ exports.rspackOptions = zod_1.z.strictObject({
     builtins: builtins.optional(),
     module: moduleOptions.optional(),
     profile: profile.optional(),
-    bail: bail.optional()
+    bail: bail.optional(),
+    features: zod_1.z.custom().optional(),
 });
 //# sourceMappingURL=zod.js.map
