diff --git a/dist/config/adapter.js b/dist/config/adapter.js
index 021b79b64d88a89baca5eebae5376e3c84438d96..ed00bf266a3c8f9a54baf40a2b03b500d88f2894 100644
--- a/dist/config/adapter.js
+++ b/dist/config/adapter.js
@@ -15,6 +15,7 @@ const getRawOptions = (options, compiler) => {
     const mode = options.mode;
     const experiments = getRawExperiments(options.experiments);
     return {
+        features: options.features,
         mode,
         target: getRawTarget(options.target),
         context: options.context,
diff --git a/dist/config/defaults.js b/dist/config/defaults.js
index 98126c146a2e38b407170113b82d070ac25c3371..464ed60cf455c6ff1be7a11061974353770346c2 100644
--- a/dist/config/defaults.js
+++ b/dist/config/defaults.js
@@ -53,6 +53,11 @@ const applyRspackOptionsDefaults = (options) => {
     applyExperimentsDefaults(options.experiments, {
         cache: options.cache
     });
+    if (options.features) {
+        applyFeaturesDefaults(options.features);
+    } else {
+        D(options, 'features', {});
+    }
     applySnapshotDefaults(options.snapshot, { production });
     applyModuleDefaults(options.module, {
         // syncWebAssembly: options.experiments.syncWebAssembly,
@@ -115,6 +120,13 @@ const applyExperimentsDefaults = (experiments, { cache }) => {
         D(experiments.rspackFuture, "disableApplyEntryLazily", true);
     }
 };
+const applyFeaturesDefaults = (features) => {
+    D(features, 'split_chunks_strategy', {});
+    if (typeof features.split_chunks_strategy === 'object') {
+        D(features.split_chunks_strategy, 'name', '');
+        D(features.split_chunks_strategy, 'topLevelFrameworks', []);
+    }
+};
 const applySnapshotDefaults = (snapshot, { production }) => {
     F(snapshot, "module", () => production
         ? { timestamp: true, hash: true }
diff --git a/dist/config/normalization.js b/dist/config/normalization.js
index e13db5ae3115aa455954784fe48be4145b94e05b..f9a53cb50c2becf65ba3233ec7dee9581d0a630a 100644
--- a/dist/config/normalization.js
+++ b/dist/config/normalization.js
@@ -12,6 +12,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.getNormalizedRspackOptions = void 0;
 const getNormalizedRspackOptions = (config) => {
     return {
+        features: config.features,
         ignoreWarnings: config.ignoreWarnings !== undefined
             ? config.ignoreWarnings.map(ignore => {
                 if (typeof ignore === "function") {
diff --git a/dist/config/zod.js b/dist/config/zod.js
index ae3c80c1e17b66c6ceeff02e06ee9d119c795d67..0c07c2044f81ec6b322fcf201ceab1fde4926187 100644
--- a/dist/config/zod.js
+++ b/dist/config/zod.js
@@ -747,5 +747,6 @@ exports.rspackOptions = zod_1.z.strictObject({
     builtins: builtins.optional(),
     module: moduleOptions.optional(),
     profile: profile.optional(),
-    bail: bail.optional()
+    bail: bail.optional(),
+    features: zod_1.z.custom().optional(),
 });
