diff --git a/dist/config/adapter.js b/dist/config/adapter.js
index 49122b3e9f4f0ac85b4075a98f693986b4333051..c3436a090fdbac00e555b2e308974f8d3f9c2426 100644
--- a/dist/config/adapter.js
+++ b/dist/config/adapter.js
@@ -15,6 +15,7 @@ const getRawOptions = (options, compiler) => {
     const mode = options.mode;
     const experiments = getRawExperiments(options.experiments);
     return {
+        features: options.features,
         mode,
         target: getRawTarget(options.target),
         context: options.context,
diff --git a/dist/config/defaults.js b/dist/config/defaults.js
index 2bc4a397888a607cb05e4eef9de7257ab8a74ba7..8cf7865d0741c14242e02901dc38c5b46b08f267 100644
--- a/dist/config/defaults.js
+++ b/dist/config/defaults.js
@@ -53,6 +53,11 @@ const applyRspackOptionsDefaults = (options) => {
     applyExperimentsDefaults(options.experiments, {
         cache: options.cache
     });
+    if (options.features) {
+        applyFeaturesDefaults(options.features);
+    } else {
+        D(options, 'features', {});
+    }
     applySnapshotDefaults(options.snapshot, { production });
     applyModuleDefaults(options.module, {
         // syncWebAssembly: options.experiments.syncWebAssembly,
@@ -120,6 +125,13 @@ const applyExperimentsDefaults = (experiments, { cache }) => {
         }
     }
 };
+const applyFeaturesDefaults = (features) => {
+    D(features, 'split_chunks_strategy', {});
+    if (typeof features.split_chunks_strategy === 'object') {
+        D(features.split_chunks_strategy, 'name', '');
+        D(features.split_chunks_strategy, 'topLevelFrameworks', []);
+    }
+};
 const applySnapshotDefaults = (snapshot, { production }) => {
     F(snapshot, "module", () => production
         ? { timestamp: true, hash: true }
diff --git a/dist/config/normalization.js b/dist/config/normalization.js
index 72167c504bd43f606481ccd1de0bdcb9eae71d9f..b16f9f3bbf1c45e00a9ca31684cefec55198683b 100644
--- a/dist/config/normalization.js
+++ b/dist/config/normalization.js
@@ -12,6 +12,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.getNormalizedRspackOptions = void 0;
 const getNormalizedRspackOptions = (config) => {
     return {
+        features: config.features,
         ignoreWarnings: config.ignoreWarnings !== undefined
             ? config.ignoreWarnings.map(ignore => {
                 if (typeof ignore === "function") {
diff --git a/dist/config/zod.js b/dist/config/zod.js
index 7a0a604ddae2eade51aae45636e16e064d8760be..6ebfcd9e9e2aebff0f95194072f87765b812f463 100644
--- a/dist/config/zod.js
+++ b/dist/config/zod.js
@@ -764,6 +764,7 @@ exports.rspackOptions = zod_1.z.strictObject({
     builtins: builtins.optional(),
     module: moduleOptions.optional(),
     profile: profile.optional(),
-    bail: bail.optional()
+    bail: bail.optional(),
+    features: zod_1.z.custom().optional(),
 });
 //# sourceMappingURL=zod.js.map
