diff --git a/dist/config/adapter.js b/dist/config/adapter.js
index 49122b3e9f4f0ac85b4075a98f693986b4333051..c3436a090fdbac00e555b2e308974f8d3f9c2426 100644
--- a/dist/config/adapter.js
+++ b/dist/config/adapter.js
@@ -15,6 +15,7 @@ const getRawOptions = (options, compiler) => {
     const mode = options.mode;
     const experiments = getRawExperiments(options.experiments);
     return {
+        features: options.features,
         mode,
         target: getRawTarget(options.target),
         context: options.context,
diff --git a/dist/config/defaults.js b/dist/config/defaults.js
index aeb7b74dd8473fa049002cf710ae860577d6d2e2..0dd4fd09fd08482b5a2c5a82cad61ab70e2b2d68 100644
--- a/dist/config/defaults.js
+++ b/dist/config/defaults.js
@@ -53,6 +53,11 @@ const applyRspackOptionsDefaults = (options) => {
     applyExperimentsDefaults(options.experiments, {
         cache: options.cache
     });
+    if (options.features) {
+        applyFeaturesDefaults(options.features);
+    } else {
+        D(options, 'features', {});
+    }
     applySnapshotDefaults(options.snapshot, { production });
     applyModuleDefaults(options.module, {
         // syncWebAssembly: options.experiments.syncWebAssembly,
@@ -103,6 +108,13 @@ const applyInfrastructureLoggingDefaults = (infrastructureLogging) => {
     D(infrastructureLogging, "colors", tty);
     D(infrastructureLogging, "appendOnly", !tty);
 };
+const applyFeaturesDefaults = (features) => {
+    D(features, 'split_chunks_strategy', {});
+    if (typeof features.split_chunks_strategy === 'object') {
+        D(features.split_chunks_strategy, 'name', '');
+        D(features.split_chunks_strategy, 'topLevelFrameworks', []);
+    }
+};
 const applyExperimentsDefaults = (experiments, { cache }) => {
     D(experiments, "lazyCompilation", false);
     D(experiments, "asyncWebAssembly", false);
diff --git a/dist/config/normalization.js b/dist/config/normalization.js
index 72167c504bd43f606481ccd1de0bdcb9eae71d9f..b16f9f3bbf1c45e00a9ca31684cefec55198683b 100644
--- a/dist/config/normalization.js
+++ b/dist/config/normalization.js
@@ -12,6 +12,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.getNormalizedRspackOptions = void 0;
 const getNormalizedRspackOptions = (config) => {
     return {
+        features: config.features,
         ignoreWarnings: config.ignoreWarnings !== undefined
             ? config.ignoreWarnings.map(ignore => {
                 if (typeof ignore === "function") {
diff --git a/dist/config/zod.js b/dist/config/zod.js
index 1c4c2cf4d02c562b8fc965c103de0eaf8e246b6a..7fa7e42a02180081096d77f64d7df0cc7d61b4d1 100644
--- a/dist/config/zod.js
+++ b/dist/config/zod.js
@@ -766,6 +766,7 @@ exports.rspackOptions = zod_1.z.strictObject({
     builtins: builtins.optional(),
     module: moduleOptions.optional(),
     profile: profile.optional(),
-    bail: bail.optional()
+    bail: bail.optional(),
+    features: zod_1.z.custom().optional(),
 });
 //# sourceMappingURL=zod.js.map
